### PR TITLE
Add easings support for library

### PR DESCRIPTION
**Pull request by Underclassity**
Add easings support for library as option.
Example:
'0 0, vh 1 | speed=2 easing="easeInQuad"'

### CODE REVIEW

**Comment by alexfoxy** [comment]
Looks good, do you have a code pen or something of it in action ?

**Comment by Underclassity** [comment]
@alexfoxy Fix some errors =)
Add easings demo page with demo usage

**Comment by sharanda** [comment]
@Underclassity thanks for great work.
It took me some time to understand why I couldn't see any effect on my side.
Turned out, it adds easing to animation progression. I'd really love have an option to add a momentum effect after a scroll event, like here: http://free.matthieu.com/jquery.parallax-scroll/demo.html

Right now I achieve the desired effect with CSS style:
`.lax { transition: transform 1.5s cubic-bezier(0,.6,.3,1) }`

Not sure about that solution. I assume it'd perform better through JS with requestAnimationFrame.

**Comment by alexfoxy** [comment]
@sharanda @Underclassity what's the feeling on this one? What is the use case for easing the animation, is it for scroll wheels for example? I can see the want for momentum after the scroll event but not sure about the easing on the animation progression?

**Comment by Underclassity** [comment]
@alexfoxy 
Easings work well on animations with a long scroll, they add physicality to the animation. At short scroll animation works almost like linear.

File: lib/lax.js [hunk 1]
```diff
@@ -30,6 +30,65 @@ function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
     var lastY = 0;
     var currentBreakpoint = 'default';
     var breakpointsSeparator = "_";
+    /*
+     * Easing Functions - inspired from http://gizma.com/easing/
+     * only considering the t value for the range [0, 1] => [0, 1]
+     */
+
+    var easings = {
```
Comment (jt3k) [inline lib/lax.js]: Why you not use a ready-made library?
e.g. https://github.com/mattdesl/eases

Comment (Underclassity) [reply lib/lax.js]: At this moment library without any dependencies, dont want to add some of them

**Comment by janein** [comment]
I totally get the desire for easingson progress. i.e. when I want to move one object towards another it should make fast progress in the beginning but then slowly get the last few steps done --> easeOutSomething

The "momentum-hack" with transition works well for the other use-case with scroll-wheel-steps. But I sometimes get into trouble because on the first animation you can see the element moving from "nothing" to the first animation step. So a js-solution would probably be nice here too.

**Comment by alexfoxy** [comment]
@Underclassity I finally have a bit of time so thinking of doing a 1.3.0 release soon. How have you been getting on with the easings? Does a CSS approach have a similar impact? This would be preferred as we don't have to increase the library size.

**Comment by sharanda** [comment]
> Does a CSS approach have a similar impact?

@alexfoxy Happy to hear you're planning further updates. Just to remind, we were talking about 2 different things. One is animation progression easing suggested by @Underclassity and second is momentum effect from current value to the new value that might be solved with a CSS transition.

Surprisingly, CSS transition approach seem to perform well overall. However, I had to manually disable it for older Microsoft Edge (animation was glitching). Furthermore, I found that Safari performs effects better with 'will-change' value. However, if your page is too long (like mine) you shouldn't have too many will-change properties. What I had to do is to trigger a class that contains 'will-change' based on user scroll, just like Apple does it on their page, for example: https://www.apple.com/iphone-se/

**Comment by alexfoxy** [comment]
@sharanda @Underclassity I created a 2.0 version of lax last weekend and wanted to your feedback if you had time. Check the branch out here: https://github.com/alexfoxy/laxxx/tree/version-2

Lots of new fun stuff :)

**Comment by sharanda** [comment]
@alexfoxy awesome news! I'll give a try this week

**Pull request closed without merging**